tests/unit/test_graph.py

```diff
@@ -126,6 +126,114 @@ def test_transfer(community_with_trustlines):
+    """A owes money to B and A wants to reduce that amount with the help of C"""
+    complex_community_with_trustlines_and_fees.update_balance(A, B, -10000)  # amount B owes A
+    complex_community_with_trustlines_and_fees.update_balance(A, C, 10000)
+    complex_community_with_trustlines_and_fees.update_balance(B, D, -10000)
+    complex_community_with_trustlines_and_fees.update_balance(C, D, 10000)
+    cost, path = complex_community_with_trustlines_and_fees.find_path_triangulation(
+            A, B, C, 10000)
+    assert path == [A, C, D, B, A]
+    assert cost == 0
+    complex_community_with_trustlines_and_fees.triangulation_transfer(A, B, C, 10000)
+    assert complex_community_with_trustlines_and_fees.get_account_sum(A, B).balance == 0
+
+
+def test_triangulation_with_cost_exact_amount(complex_community_with_trustlines_and_fees):
+    """A owes money to B and A wants to reduce that amount with the help of C"""
+    complex_community_with_trustlines_and_fees.update_balance(A, B, -10000)  # amount B owes A
+    complex_community_with_trustlines_and_fees.update_balance(A, C, -10000)
+    complex_community_with_trustlines_and_fees.update_balance(B, D, 10000)
+    complex_community_with_trustlines_and_fees.update_balance(C, D, -10000)
+    cost, path = complex_community_with_trustlines_and_fees.find_path_triangulation(
+            A, B, C, 10000)
+    assert path == [A, C, D, B, A]
+    assert cost == 306
+    complex_community_with_trustlines_and_fees.triangulation_transfer(A, B, C, 10000)
+    assert complex_community_with_trustlines_and_fees.get_account_sum(A, B).balance == 0
+
+
+def test_triangulation_not_exact_amount(complex_community_with_trustlines_and_fees):
+    """A owes money to B and A wants to reduce that amount with the help of C"""
+    complex_community_with_trustlines_and_fees.update_balance(A, B, -10000)  # amount B owes A
+    complex_community_with_trustlines_and_fees.update_balance(A, C, 10000)
+    complex_community_with_trustlines_and_fees.update_balance(B, D, -10000)
+    complex_community_with_trustlines_and_fees.update_balance(C, D, 10000)
+    cost, path = complex_community_with_trustlines_and_fees.find_path_triangulation(
+            A, B, C, 1000)
+    assert path == [A, C, D, B, A]
+    assert cost == 0
+    complex_community_with_trustlines_and_fees.triangulation_transfer(A, B, C, 1000)
+    assert complex_community_with_trustlines_and_fees.get_account_sum(A, B).balance == -9000
+
+
+def test_triangulation_not_enough_capacity(complex_community_with_trustlines_and_fees):
+    """A owes money to B and A wants to reduce that amount with the help of C"""
+    complex_community_with_trustlines_and_fees.update_balance(A, B, -10000)  # amount B owes A
+    complex_community_with_trustlines_and_fees.update_balance(A, C, 10000)
+    complex_community_with_trustlines_and_fees.update_balance(B, D, -10000)
+    complex_community_with_trustlines_and_fees.update_balance(C, D, 10000)
+    with pytest.raises(ValueError):
```

Comment:
This should not raise ValueError but gives a empty path like it is now in all other cases when there is no path.

Or we change it that an exception is thrown when there is no path (would be more pythonic, we could use a default parameter to suppress the exception how it is for example in a dict when looking for a value), but then we want to have an own Exception and it needs to be thrown every time when there is now path, so also in findPath. And this needs to be documented then.